cleanup/tests/test_daemon.py

```
@@ -0,0 +1,392 @@
+        daemon_mod._start_idle_exiter = orig_start
+
+
+def test_cli_warmup_flag_against_running_daemon():
+    """End-to-end: CLI --warmup hits the daemon and exits 0."""
+    with tempfile.TemporaryDirectory() as td:
+        sock = str(Path(td) / "d.sock")
+        engine = FakeEngine("fake")
+        engine.load_delay_s = 0.3
+        thread = _serve_in_thread(sock, engine, idle_exit_seconds=3600)
+        try:
+            r = subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)